play-json/shared/src/main/scala-3/play/api/libs/json/JsMacroImpl.scala

```diff
@@ -147,4 +145,12 @@ object JsMacroImpl {
     inline val url = "https://github.com/lampepfl/dotty/issues/" + issueNumber
     error(msg + ", see " + url)
   }
+
+  import scala.quoted._
+  private def summonClassNameImpl[A: Type](using Quotes): Expr[String] = {
+    import quotes.reflect._
+    val sym  = TypeRepr.of[A].typeSymbol
+    val fqcn = sym.owner.fullName.stripSuffix("$") + "." + sym.name
```

Comment:
```suggestion
    val fqcn = sym.owner.fullName.stripSuffix("$") + "." + sym.name.stripSuffix("$")
```
this should fix the tests